### PR TITLE
Explorer storage s3

### PR DESCRIPTION
**Pull request by flicker-harmony**
## Issue

Same thing as for master. Made fixes.

## Test

Screenshot 2020-02-15 at 18 56 02

### Unit Test Coverage

Before:

```
<!-- copy/paste 'go test -cover' result in the directory you made change -->
```

After:

```
<!-- copy/paste 'go test -cover' result in the directory you made change -->
```

## Operational Checklist

1. **Does this PR introduce backward-incompatible changes to the on-disk data structure and/or the over-the-wire protocol?**. (If no, skip to question 8.)

    **YES|NO**

2. **Describe the migration plan.**. For each flag epoch, describe what changes take place at the flag epoch, the anticipated interactions between upgraded/non-upgraded nodes, and any special operational considerations for the migration.

3. **Describe how the plan was tested.**

4. **How much minimum baking period after the last flag epoch should we allow on Pangaea before promotion onto mainnet?**

5. **What are the planned flag epoch numbers and their ETAs on Pangaea?**

6. **What are the planned flag epoch numbers and their ETAs on mainnet?**

    Note that this must be enough to cover baking period on Pangaea.

7. **What should node operators know about this planned change?**

8. **Does this PR introduce backward-incompatible changes *NOT* related to on-disk data structure and/or over-the-wire protocol?** (If no, continue to question 11.)

    **YES|NO**

9. **Does the existing `node.sh` continue to work with this change?**

10. **What should node operators know about this change?**

11. **Does this PR introduce significant changes to the operational requirements of the node software, such as >20% increase in CPU, memory, and/or disk usage?**

### CODE REVIEW

File: api/service/explorer/storage.go [hunk 1]
```diff
@@ -74,85 +49,44 @@ func (storage *Storage) Init(ip, port string, remove bool) {
 	if remove {
 		var err = os.RemoveAll(dbFileName)
 		if err != nil {
-			utils.Logger().Error().Err(err).Msg("Failed to remove existing database files")
+			utils.Logger().Error().Err(err).Msg("Failed to re	move existing database files")
```
Comment (LeoHChen) [inline api/service/explorer/storage.go]: ??

File: api/service/explorer/storage.go [hunk 2]
```diff
@@ -74,85 +49,44 @@ func (storage *Storage) Init(ip, port string, remove bool) {
...
 		}
 	}
-	if storage.db, err = ethdb.NewLDBDatabase(dbFileName, 0, 0); err != nil {
+	options := &opt.Options{
+		OpenFilesCacheCapacity: 0,
```
Comment (LeoHChen) [inline api/service/explorer/storage.go]: what's the meaning of those options here?  Can you add some more comments?

File: api/service/explorer/storage.go (hunk 1 above)
Comment (flicker-harmony) [reply api/service/explorer/storage.go]: Ok

File: api/service/explorer/storage.go (hunk 2 above)
Comment (flicker-harmony) [reply api/service/explorer/storage.go]: Those are standard options for ethdb how it was created from leveldb.

**Comment by flicker-harmony** [comment]
@LeoHChen no, we don't need to upgrade core database, its used only for explorer storage. I did that to have proper iterator range method. But its essentially same leveldb created with the same params, I checked. Considering that newer leveldb versions should be compatible with old ones this one should be okay, but needs to be tried on existing testnet.

**Comment by janet-harmony** [comment]
> LGTM in general, can you work with Janet to test this patch on one of the explorer node in LRTN?

Deployed on LRTN S2 explorer node this morning. Shard 2 on block explorer looks fine.
Saw this error in the node logs, but the node is syncing correctly.
```
$ grep error zerolog-validator-54.84.188.10-9000.log
{"level":"error","port":"9000","ip":"54.84.188.10","caller":"/home/ec2-user/JL/harmony/core/blockchain.go:1673","time":"2020-02-26T20:51:00.939695798Z","message":"\n########## BAD BLOCK #########\nChain config: {ChainID: 2 EIP155: 0 CrossTx: 1 CrossLink: 2 ReceiptLog: 0}\n\nNumber: 369705\nHash: 0x0c564007020fbdba1efcb4020a8a290a5ce93eeb27c43f2085e822ac7f208cd2\n\n\nError: unknown ancestor\n##############################\n"}
{"level":"error","port":"9000","ip":"54.84.188.10","error":"unknown ancestor","blockNum":369705,"parentHash":"0x348486e0813efaa0321b349b4fdaa2a18e78773baddc0cee7876315795a11e66","hash":"0x0c564007020fbdba1efcb4020a8a290a5ce93eeb27c43f2085e822ac7f208cd2","caller":"/home/ec2-user/JL/harmony/node/node_handler.go:445","time":"2020-02-26T20:51:00.939724287Z","message":"Error Adding new block to blockchain"}
{"level":"error","port":"9000","ip":"54.84.188.10","error":"unknown ancestor","caller":"/home/ec2-user/JL/harmony/node/node_explorer.go:129","time":"2020-02-26T20:51:00.939735905Z","message":"[Explorer] Error when adding new block for explorer node"}
```
Return from `hmy_getTransactionHistory`. Continues for 352 transactions, same as displayed on explorer.
Screen Shot 2020-02-26 at 7 01 21 PM

**Comment by LeoHChen** [comment]
@flicker-harmony , can you check the /address on the new end point?

**Pull request merged**